consensus/qbft/src/integration-test/java/org/hyperledger/besu/consensus/qbft/test/LocalNodeNotProposerTest.java

```diff
@@ -0,0 +1,129 @@
+  private final RoundSpecificPeers peers = context.roundSpecificPeers(roundId);
+
+  private final MessageFactory localNodeMessageFactory = context.getLocalNodeMessageFactory();
+
+  private final Block blockToPropose = context.createBlockForProposalFromChainHead(0, 15);
+
+  private Prepare expectedTxPrepare;
+  private Commit expectedTxCommit;
+
+  @Before
+  public void setup() {
+    expectedTxPrepare = localNodeMessageFactory.createPrepare(roundId, blockToPropose.getHash());
+
+    expectedTxCommit =
+        new Commit(
+            createSignedCommitPayload(
+                roundId, blockToPropose, context.getLocalNodeParams().getNodeKey()));
+  }
+
+  @Test
+  public void preparesReceivedFromNonProposerIsValid() {
+    peers.getProposer().injectProposal(roundId, blockToPropose);
+
+    peers.verifyMessagesReceived(expectedTxPrepare);
+
+    peers.getNonProposing(0).injectPrepare(roundId, blockToPropose.getHash());
+    peers.verifyMessagesReceived(expectedTxCommit);
+
+    // Ensure the local blockchain has NOT incremented yet.
+    assertThat(context.getCurrentChainHeight()).isEqualTo(0);
+
+    // NO further messages should be transmitted when another Prepare is received.
+    peers.getNonProposing(1).injectPrepare(roundId, blockToPropose.getHash());
+    peers.verifyNoMessagesReceived();
+
+    // Inject a commit, ensure blockChain is not updated, and no message are sent (not quorum yet)
+    peers.getNonProposing(0).injectCommit(roundId, blockToPropose.getHash());
+    peers.verifyNoMessagesReceived();
+    assertThat(context.getCurrentChainHeight()).isEqualTo(0);
+
+    // A second commit message means quorum is reached, and blockchain should be updated.
+    peers.getNonProposing(1).injectCommit(roundId, blockToPropose.getHash());
+    peers.verifyNoMessagesReceived();
+    assertThat(context.getCurrentChainHeight()).isEqualTo(1);
+
+    // ensure any further commit messages do not affect the system
+    peers.getProposer().injectCommit(roundId, blockToPropose.getHash());
+    peers.verifyNoMessagesReceived();
+    assertThat(context.getCurrentChainHeight()).isEqualTo(1);
+  }
+
+  @Test
+  public void commitMessagesReceivedBeforePrepareCorrectlyImports() {
+    // All peers send a commit, then all non-proposing peers send a prepare, when then Proposal
+    // arrives last, the chain is updated, and a prepare and commit message are transmitted.
+    peers.clearReceivedMessages();
+    peers.commit(roundId, blockToPropose.getHash());
+    peers.verifyNoMessagesReceived();
+    assertThat(context.getCurrentChainHeight()).isEqualTo(0);
+
+    peers.prepareForNonProposing(roundId, blockToPropose.getHash());
+    peers.verifyNoMessagesReceived();
+    assertThat(context.getCurrentChainHeight()).isEqualTo(0);
+
+    peers.getProposer().injectProposal(roundId, blockToPropose);
+    // TODO(tmm): Unfortunately, there are times that the Commit will go out BEFORE the prepare
```

Comment:
nit: TODO meant to be removed?

Reply:
no unfortunately. this hasn't been fixed yet